x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -296,24 +297,29 @@ func getMetricDataPerRegion(durationString string, periodInSec int, instanceID s
 	return getMetricDataOutput, nil
 }
 
-func createMetricDataQuery(id string, metricName string, periodInSec int, dimensions []cloudwatch.Dimension) (metricDataQuery cloudwatch.MetricDataQuery) {
-	namespace := "AWS/EC2"
+func createMetricDataQuery(metric cloudwatch.Metric, instanceID string, index int, periodInSec int) (metricDataQuery cloudwatch.MetricDataQuery) {
 	statistic := "Average"
 	period := int64(periodInSec)
-
-	metric := cloudwatch.Metric{
-		Namespace:  &namespace,
-		MetricName: &metricName,
-		Dimensions: dimensions,
-	}
-
-	metricDataQuery = cloudwatch.MetricDataQuery{
-		Id: &id,
-		MetricStat: &cloudwatch.MetricStat{
-			Period: &period,
-			Stat:   &statistic,
-			Metric: &metric,
-		},
+	id := "e" + strconv.Itoa(index)
+	metricDims := metric.Dimensions
+
+	for _, dim := range metricDims {
+		if *dim.Name == "InstanceId" {
+			if instanceID == *dim.Value {
```

Comment:
For readability I'd reduce nesting levels here with:
```
if *dim.Name != "InstanceID" || instanceID != *dim.Value {
    continue
}
```

Reply:
ha I think `*dim.Name == "InstanceId" && *dim.Value == instanceID` is more readable in this case :-)